sdk/python/kfp/compiler/test_data/pipelines/pipeline_with_various_io_types.py

```diff
@@ -15,13 +15,15 @@
 from kfp import compiler
 from kfp import components
 from kfp import dsl
+from kfp.dsl import Artifact
+from kfp.dsl import Input
 
 component_op_1 = components.load_component_from_text("""
 name: upstream
 inputs:
 - {name: input_1, type: String}
 - {name: input_2, type: Float}
-- {name: input_3, type: String}
+- {name: input_3, type: Artifact}
```

Comment:
I think this is a good test of the compilation logic, but this is not a valid pipeline (cannot be run after compiled and submitted).

When submitting a pipeline, do you think that it makes sense to validate in the `Client` that the outermost pipeline does not have an artifact as an input?

Reply:
No, this will become a valid pipeline. There will be client changes to support submitting such a pipeline with artifact "value" provided via `runtimeConfig`. Details can be found in Rui's design doc.